software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java

```diff
@@ -336,7 +336,7 @@ protected int tryJavaInstall(String version, String command) {
         log.debug("Checking Java version at {}@{}", getEntity(), getLocation());
         // sed gets stdin like 'java version "1.7.0_45"'
         ProcessTaskWrapper<Integer> versionCommand = Entities.submit(getEntity(), SshTasks.newSshExecTaskFactory(
-                getLocation(), "java -version 2>&1 | grep \"java version\" | sed 's/.*\"\\(.*\\).*\"/\\1/'"));
+                getLocation(), "java -version 2>&1 | grep \" version\" | sed 's/.*\"\\(.*\\).*\"/\\1/'"));
```

Comment:
not sure how universal `grep -E` is so i'd be against that.  can you add a mention to `openjdk version "1.8.0_45"` in the comment?  things get missed in the git history, especially if not on the first line (although they shouldn't -- apologies from me for not noticing it!)